### PR TITLE
torchci: Add ability to check nightly for alerts

### PR DESCRIPTION
**Pull request by seemethere**
Adds the ability to check nightly for alerting as well as create separate issues for separate branches.

Also makes it so that this script isn't hardcoded to 'pytorch/pytorch' and adds a dry-run mode so that we can skip doing real work if you're just developing this script.

Base line functionality shouldn't change with this aside from the following:

* Removed code blocks when posting the update comment and instead replaced it with a link to hud (minihud) so that it's clickable

After this we'll need to create an internal butterfly rule to manage these internally.

<details>
<summary> Click me for dry run details </summary>

```
❯ python3 check_alerts.py --dry-run --repo pytorch/pytorch --branch nightly
The first green SHA was at index -1 at d80585d52dba17745a8eecb4dbdbfdcba5f254bband the first red SHA was at index 0 at 6f611c50305244c6f365e123c49fb4e5ce7eadf0
Generating alerts for:  [jobName: Build Official Docker Images / build (devel, linux/amd64), jobName: Build Official Docker Images / build (runtime, linux/arm64,linux/amd64), jobName: binary_ios_upload-1 / build, jobName: linux-binary-conda / conda-py3_10-cuda11_6-build / build, jobName: linux-binary-conda / conda-py3_10-cuda11_7-build / build, jobName: linux-binary-conda / conda-py3_10-cuda11_8-build / build, jobName: linux-binary-conda / conda-py3_7-cuda11_6-build / build, jobName: linux-binary-conda / conda-py3_7-cuda11_7-build / build, jobName: linux-binary-conda / conda-py3_7-cuda11_8-build / build, jobName: linux-binary-conda / conda-py3_8-cuda11_6-build / build, jobName: linux-binary-conda / conda-py3_8-cuda11_7-build / build, jobName: linux-binary-conda / conda-py3_8-cuda11_8-build / build, jobName: linux-binary-conda / conda-py3_9-cuda11_6-build / build, jobName: linux-binary-conda / conda-py3_9-cuda11_7-build / build, jobName: linux-binary-conda / conda-py3_9-cuda11_8-build / build, jobName: linux-binary-libtorch-cxx11-abi / libtorch-cuda11_6-shared-with-deps-cxx11-abi-build / build, jobName: linux-binary-libtorch-cxx11-abi / libtorch-cuda11_6-static-with-deps-cxx11-abi-build / build, jobName: linux-binary-libtorch-cxx11-abi / libtorch-rocm5_3-shared-with-deps-cxx11-abi-test, jobName: linux-binary-libtorch-cxx11-abi / libtorch-rocm5_3-static-with-deps-cxx11-abi-test, jobName: linux-binary-libtorch-pre-cxx11 / libtorch-cuda11_6-shared-with-deps-pre-cxx11-build / build, jobName: linux-binary-libtorch-pre-cxx11 / libtorch-cuda11_6-static-with-deps-pre-cxx11-build / build, jobName: windows-binary-libtorch-debug / libtorch-cuda11_8-shared-with-deps-debug-build, jobName: windows-binary-libtorch-debug / libtorch-cuda11_8-shared-without-deps-debug-build, jobName: windows-binary-libtorch-debug / libtorch-cuda11_8-static-with-deps-debug-build, jobName: windows-binary-libtorch-debug / libtorch-cuda11_8-static-without-deps-debug-build]
Creating issue with content:
{'title': '[Pytorch] There are 25 Recurrently Failing Jobs on pytorch/pytorch nightly', 'body': 'Within the last 50 commits, there are the following failures on the master branch of pytorch: \n- [Build Official Docker Images / build (devel, linux/amd64)](https://hud.pytorch.org/minihud?name_filter=Build%20Official%20Docker%20Images%20/%20build%20%28devel%2C%20linux/amd64%29) failed consecutively starting with commit [7277d5a8f088683be4265aa36ce28b96316a8009](https://hud.pytorch.org/commit/pytorch/pytorch/7277d5a8f088683be4265aa36ce28b96316a8009)\n\n- [Build Official Docker Images / build (runtime, linux/arm64,linux/amd64)](https://hud.pytorch.org/minihud?name_filter=Build%20Official%20Docker%20Images%20/%20build%20%28runtime%2C%20linux/arm64%2Clinux/amd64%29) failed consecutively starting with commit [7277d5a8f088683be4265aa36ce28b96316a8009](https://hud.pytorch.org/commit/pytorch/pytorch/7277d5a8f088683be4265aa36ce28b96316a8009)\n\n- [binary_ios_upload-1 / build](https://hud.pytorch.org/minihud?name_filter=binary_ios_upload-1%20/%20build) failed consecutively starting with commit [d80585d52dba17745a8eecb4dbdbfdcba5f254bb](https://hud.pytorch.org/commit/pytorch/pytorch/d80585d52dba17745a8eecb4dbdbfdcba5f254bb)\n\n- [linux-binary-conda / conda-py3_10-cuda11_6-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_10-cuda11_6-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_10-cuda11_7-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_10-cuda11_7-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_10-cuda11_8-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_10-cuda11_8-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_7-cuda11_6-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_7-cuda11_6-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_7-cuda11_7-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_7-cuda11_7-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_7-cuda11_8-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_7-cuda11_8-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_8-cuda11_6-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_8-cuda11_6-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_8-cuda11_7-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_8-cuda11_7-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_8-cuda11_8-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_8-cuda11_8-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_9-cuda11_6-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_9-cuda11_6-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_9-cuda11_7-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_9-cuda11_7-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-conda / conda-py3_9-cuda11_8-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-conda%20/%20conda-py3_9-cuda11_8-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-libtorch-cxx11-abi / libtorch-cuda11_6-shared-with-deps-cxx11-abi-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-libtorch-cxx11-abi%20/%20libtorch-cuda11_6-shared-with-deps-cxx11-abi-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-libtorch-cxx11-abi / libtorch-cuda11_6-static-with-deps-cxx11-abi-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-libtorch-cxx11-abi%20/%20libtorch-cuda11_6-static-with-deps-cxx11-abi-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-libtorch-cxx11-abi / libtorch-rocm5_3-shared-with-deps-cxx11-abi-test](https://hud.pytorch.org/minihud?name_filter=linux-binary-libtorch-cxx11-abi%20/%20libtorch-rocm5_3-shared-with-deps-cxx11-abi-test) failed consecutively starting with commit [06f81fdbf8817cd986f77ec2c67f50489a40ecf9](https://hud.pytorch.org/commit/pytorch/pytorch/06f81fdbf8817cd986f77ec2c67f50489a40ecf9)\n\n- [linux-binary-libtorch-cxx11-abi / libtorch-rocm5_3-static-with-deps-cxx11-abi-test](https://hud.pytorch.org/minihud?name_filter=linux-binary-libtorch-cxx11-abi%20/%20libtorch-rocm5_3-static-with-deps-cxx11-abi-test) failed consecutively starting with commit [06f81fdbf8817cd986f77ec2c67f50489a40ecf9](https://hud.pytorch.org/commit/pytorch/pytorch/06f81fdbf8817cd986f77ec2c67f50489a40ecf9)\n\n- [linux-binary-libtorch-pre-cxx11 / libtorch-cuda11_6-shared-with-deps-pre-cxx11-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-libtorch-pre-cxx11%20/%20libtorch-cuda11_6-shared-with-deps-pre-cxx11-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [linux-binary-libtorch-pre-cxx11 / libtorch-cuda11_6-static-with-deps-pre-cxx11-build / build](https://hud.pytorch.org/minihud?name_filter=linux-binary-libtorch-pre-cxx11%20/%20libtorch-cuda11_6-static-with-deps-pre-cxx11-build%20/%20build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [windows-binary-libtorch-debug / libtorch-cuda11_8-shared-with-deps-debug-build](https://hud.pytorch.org/minihud?name_filter=windows-binary-libtorch-debug%20/%20libtorch-cuda11_8-shared-with-deps-debug-build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [windows-binary-libtorch-debug / libtorch-cuda11_8-shared-without-deps-debug-build](https://hud.pytorch.org/minihud?name_filter=windows-binary-libtorch-debug%20/%20libtorch-cuda11_8-shared-without-deps-debug-build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [windows-binary-libtorch-debug / libtorch-cuda11_8-static-with-deps-debug-build](https://hud.pytorch.org/minihud?name_filter=windows-binary-libtorch-debug%20/%20libtorch-cuda11_8-static-with-deps-debug-build) failed consecutively starting with commit [f960e5c8ae2482a03a3f525457a922e9209bc00d](https://hud.pytorch.org/commit/pytorch/pytorch/f960e5c8ae2482a03a3f525457a922e9209bc00d)\n\n- [windows-binary-libtorch-debug / libtorch-cuda11_8-static-without-deps-debug-build](https://hud.pytorch.org/minihud?name_filter=windows-binary-libtorch-debug%20/%20libtorch-cuda11_8-static-without-deps-debug-build) failed consecutively starting with commit [284677213e9dab78d8d01c39aa0e94faf67bd029](https://hud.pytorch.org/commit/pytorch/pytorch/284677213e9dab78d8d01c39aa0e94faf67bd029)\n\nPlease review the errors and revert if needed.', 'labels': ['pytorch-alert']}
NOTE: Dry run activated, not doing any real work
```
</details>

Signed-off-by: Eli Uriegas <eliuriegas@meta.com>

### CODE REVIEW

File: torchci/scripts/check_alerts.py [hunk 1]
```diff
@@ -246,15 +252,12 @@ def gen_update_comment(original_body: str, jobs: List[JobStatus]) -> str:
...
-            s += f"* {job}\n"
+            s += f"* {generate_failed_job_hud_link(job)}\n"
         s += "\n"
     if len(started_failing_jobs) > 0:
         s += "These jobs started failing:\n"
         for job in started_failing_jobs:
-            s += f"* {job}\n"
-    if s:
-        # funky string to make internal alert messaging have code blocks
-        return f"````\n```\n{s}\n```\n````"
```
Comment (seemethere) [inline torchci/scripts/check_alerts.py:257]: cc @clee2000 since you were the one who originally added code blocks here, is it okay to remove these?

**Review by malfet — APPROVED** [review]

**Review by atalman — APPROVED** [review]
LGTM

**Comment by seemethere** [comment]
Interestingly enough when I attempt to run this for other repositories it starts to fail out with a 500 Server Error:

```
 ❯ REPO_TO_CHECK=pytorch/torchaudio BRANCH_TO_CHECK=nightly python3 check_alerts.py
Traceback (most recent call last):
  File "/Users/eliuriegas/work/test-infra/torchci/scripts/check_alerts.py", line 524, in <module>
    main()
  File "/Users/eliuriegas/work/test-infra/torchci/scripts/check_alerts.py", line 517, in main
    check_for_recurrently_failing_jobs_alert(args.repo, args.branch, args.dry_run)
  File "/Users/eliuriegas/work/test-infra/torchci/scripts/check_alerts.py", line 448, in check_for_recurrently_failing_jobs_alert
    job_names, sha_grid = fetch_hud_data(repo=repo, branch=branch)
  File "/Users/eliuriegas/work/test-infra/torchci/scripts/check_alerts.py", line 309, in fetch_hud_data
    response.raise_for_status()
  File "/Users/eliuriegas/brew/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://hud.pytorch.org/api/hud/pytorch/torchaudio/nightly/0
```

@ZainRizvi @huydhn do you have any idea why this might be?

**Comment by ZainRizvi** [comment]
@seemethere that's because the repo name is audio, not torchaudio :) 

This link works just fine:
https://hud.pytorch.org/api/hud/pytorch/audio/main/1

**Pull request merged**